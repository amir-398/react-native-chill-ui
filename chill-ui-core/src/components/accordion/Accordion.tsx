import { cn } from '@utils';

import type { AccordionProps } from '../../types/accordion.types';

import { Box } from '../box';
import styles from './Accordion.style';
import { AccordionProvider } from './AccordionContext';
import { isNativeWindInstalled } from '../../utils/hybrid/nativewindDetector';
import { classNamePropsHandler } from '../../utils/hybrid/classNamePropsHandler';

/**
 * Accordion component provides a collapsible content area with support for single or multiple items.
 *
 * @example
 * ```tsx
 * <Accordion type="single" collapsible>
 *   <AccordionItem value="item-1">
 *     <AccordionTrigger>Is it accessible?</AccordionTrigger>
 *     <AccordionContent>Yes. It adheres to the WAI-ARIA design pattern.</AccordionContent>
 *   </AccordionItem>
 * </Accordion>
 * ```
 *
 * @param children - Child AccordionItem components
 * @param type - Whether single or multiple items can be open at once ('single' | 'multiple')
 * @param className - Custom CSS classes for styling
 * @param collapseIcon - Icon to show when item is expanded (default: 'angle-down-solid')
 * @param collapsible - Whether items can be collapsed when type is 'single' (default: false)
 * @param defaultValue - Default open items (string for single, array for multiple)
 * @param disabled - Disable the entire accordion (default: false)
 * @param expandIcon - Icon to show when item is collapsed (default: 'angle-down-solid')
 * @param hasCollapseIcon - Whether to show collapse/expand icons (default: true)
 * @param iconPosition - Position of the expand/collapse icon (default: 'right')
 * @param onValueChange - Callback fired when accordion state changes
 * @param props - Additional ViewProps for the container
 *
 * @see {@link https://github.com/your-repo/chill-ui/tree/main/src/components/accordion/README.md Documentation}
 */
export default function Accordion(props: AccordionProps) {
  classNamePropsHandler(props, 'Accordion');
  const {
    children,
    className,
    collapseIcon = 'angle-down-solid',
    collapsible = false,
    defaultValue,
    disabled = false,
    expandIcon = 'angle-down-solid',
    hasCollapseIcon = true,
    iconPosition = 'right',
    onValueChange,
    style,
    type,
  } = props;

  return (
    <AccordionProvider
      type={type}
      collapsible={collapsible}
      defaultValue={defaultValue}
      disabled={disabled}
      hasAnimation={false}
      animationDuration={0}
      hasCollapseIcon={hasCollapseIcon ?? true}
      iconPosition={iconPosition}
      expandIcon={expandIcon}
      collapseIcon={collapseIcon}
      onValueChange={onValueChange}
    >
      {isNativeWindInstalled() ? (
        <Box className={cn('w-full overflow-hidden rounded-lg', className)} style={style}>
          {children}
        </Box>
      ) : (
        <Box style={[styles.accordion, style]}>{children}</Box>
      )}
    </AccordionProvider>
  );
}
