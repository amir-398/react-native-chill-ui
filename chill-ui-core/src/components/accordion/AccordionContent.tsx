import { cn } from '@utils';

import type { AccordionContentProps } from '../../types/accordion.types';

import { Box } from '../box';
import { String } from '../string';
import styles from './Accordion.style';
import { useAccordion } from './AccordionContext';
import { isNativeWindInstalled } from '../../utils';
import { useAccordionItem } from './AccordionItemContext';
import { classNamePropsHandler } from '../../utils/hybrid/classNamePropsHandler';

/**
 * AccordionContent is the collapsible content area that shows/hides based on the accordion state.
 * Supports smooth animations and custom styling.
 *
 * @example
 * ```tsx
 * <AccordionContent stringProps={{ variant: 'body-2' }}>
 *   This content will be shown when the accordion item is expanded.
 * </AccordionContent>
 *
 * // Custom styling
 * <AccordionContent className="bg-gray-100 p-4">
 *   <Box>
 *     <String>Custom styled content</String>
 *   </Box>
 * </AccordionContent>
 * ```
 *
 * @param children - Content to display when expanded
 * @param className - Custom CSS classes
 * @param stringProps - Props to pass to String component when children is a string
 * @param ...rest - Props to pass to View props
 */
export default function AccordionContent(props: AccordionContentProps) {
  classNamePropsHandler(props, 'AccordionContent');
  const { children, className, stringProps, style, ...rest } = props;
  const { isItemOpen } = useAccordion();
  const { value } = useAccordionItem();

  const isOpen = isItemOpen(value);

  const staticHeight = isOpen ? 'auto' : 0;

  const contentStyle = isOpen ? undefined : styles.accordionContentHidden;

  return (
    <Box
      style={{
        height: staticHeight,
        overflow: 'hidden',
      }}
    >
      {isNativeWindInstalled() ? (
        <Box className={cn('bg-[#ebebeb] px-4 py-3', className)} style={style} {...rest}>
          {typeof children === 'string' ? <String {...stringProps}>{children}</String> : children}
        </Box>
      ) : (
        <Box style={[styles.accordionContent, contentStyle, style]} {...rest}>
          {typeof children === 'string' ? <String {...stringProps}>{children}</String> : children}
        </Box>
      )}
    </Box>
  );
}
