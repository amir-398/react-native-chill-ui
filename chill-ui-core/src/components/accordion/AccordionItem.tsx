import type { AccordionItemProps } from '../../types/accordion.types';

import { Box } from '../box';
import styles from './Accordion.style';
import { AccordionItemProvider } from './AccordionItemContext';
import { isNativeWindInstalled } from '../../utils/hybrid/nativewindDetector';
import { classNamePropsHandler } from '../../utils/hybrid/classNamePropsHandler';

/**
 * AccordionItem wraps a single accordion item with its trigger and content.
 *
 * @example
 * ```tsx
 * <AccordionItem value="item-1" disabled={false}>
 *   <AccordionTrigger>Question</AccordionTrigger>
 *   <AccordionContent>Answer</AccordionContent>
 * </AccordionItem>
 * ```
 *
 * @param value - Unique identifier for the item
 * @param disabled - Whether this specific item is disabled
 * @param className - Custom CSS classes
 */
export default function AccordionItem(props: AccordionItemProps) {
  const { children, className, disabled, style, value, ...rest } = props;
  classNamePropsHandler(props, 'AccordionItem');

  return (
    <AccordionItemProvider value={value} disabled={disabled}>
      {isNativeWindInstalled() ? (
        <Box className={className} style={style} {...rest}>
          {children}
        </Box>
      ) : (
        <Box style={[styles.accordionItem, style]} {...rest}>
          {children}
        </Box>
      )}
    </AccordionItemProvider>
  );
}
