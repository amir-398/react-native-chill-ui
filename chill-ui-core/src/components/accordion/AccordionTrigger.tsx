import { cn } from '@utils';
import { useState, useEffect, cloneElement, isValidElement, Children } from 'react';
import { TouchableOpacity, Animated, TouchableHighlight, Pressable } from 'react-native';

import type { AccordionTriggerProps } from '../../types/accordion.types';

import Icon from '../icon';
import { Box } from '../box';
import { String } from '../string';
import styles from './Accordion.style';
import { AnimatedBox } from '../animatedBox';
import { useAccordion } from './AccordionContext';
import { RipplePressable } from '../ripplePressable';
import { useAccordionItem } from './AccordionItemContext';
import { isNativeWindInstalled } from '../../utils/hybrid/nativewindDetector';
import { classNamePropsHandler } from '../../utils/hybrid/classNamePropsHandler';

/**
 * AccordionTrigger is the clickable header that toggles the accordion item.
 * Supports different trigger types (TouchableOpacity, Pressable, etc.) and custom styling.
 *
 * @example
 * ```tsx
 * <AccordionTrigger as="TouchableOpacity">
 *   Is it accessible?
 * </AccordionTrigger>
 *
 * // Custom content
 * <AccordionTrigger>
 *   <Box className="flex-row items-center">
 *     <Icon name="user" />
 *     <String>Custom Trigger</String>
 *   </Box>
 * </AccordionTrigger>
 *
 * // Using asChild to pass props to child component
 * <AccordionTrigger asChild>
 *   <TouchableOpacity>
 *     <String>Custom Button</String>
 *   </TouchableOpacity>
 * </AccordionTrigger>
 * ```
 *
 * @param as - Component type to use for the trigger (default: TouchableOpacity)
 * @param asChild - Use the child component as the trigger element instead of wrapping it
 * @param children - Content to display in the trigger
 * @param className - Custom CSS classes
 * @param stringProps - Props to pass to String component when children is a string
 * @param props - Additional TouchableOpacityProps
 */
export default function AccordionTrigger(props: AccordionTriggerProps) {
  classNamePropsHandler(props, 'AccordionTrigger');
  const { as, asChild, children, className, stringProps, style } = props;
  const {
    animationDuration,
    collapseIcon,
    disabled: accordionDisabled,
    expandIcon,
    hasCollapseIcon,
    iconPosition,
    isItemOpen,
    toggleItem,
  } = useAccordion();
  const { disabled: itemDisabled, value } = useAccordionItem();

  const isOpen = isItemOpen(value);

  const disabled = accordionDisabled || itemDisabled;

  const [animation] = useState(new Animated.Value(isOpen ? 1 : 0));

  useEffect(() => {
    Animated.timing(animation, {
      duration: animationDuration,
      toValue: isOpen ? 1 : 0,
      useNativeDriver: false,
    }).start();
  }, [isOpen, animationDuration, animation]);

  const animatedRotation = animation.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '180deg'],
  });

  const handlePress = () => {
    if (!disabled) {
      toggleItem(value);
    }
  };

  const commonProps = isNativeWindInstalled()
    ? {
        className: cn(
          'w-full flex-row items-center justify-between border-y border-[#E5E7EB] bg-[#F9FAFB] px-4 py-3',
          {
            'opacity-50': disabled,
          },
          className,
        ),
        disabled,
        onPress: handlePress,
        style,
        ...props,
      }
    : {
        disabled,
        onPress: handlePress,
        style: [styles.accordionTrigger, disabled && styles.accordionTriggerDisabled, style],
        ...props,
      };

  const renderContent = () => (
    <>
      {hasCollapseIcon && iconPosition === 'left' && (
        <AnimatedBox
          style={{
            transform: [{ rotate: animatedRotation }],
          }}
        >
          <Icon
            name={isOpen ? (collapseIcon as any) || 'angle-down-solid' : (expandIcon as any) || 'angle-down-solid'}
          />
        </AnimatedBox>
      )}

      <Box style={styles.accordionTriggerContent}>
        {typeof children === 'string' ? (
          <String
            {...stringProps}
            {...(isNativeWindInstalled()
              ? {
                  className: cn('text-gray-900', {
                    'ml-3': hasCollapseIcon && iconPosition === 'left',
                    'mr-3': hasCollapseIcon && iconPosition === 'right',
                  }),
                }
              : {
                  style: [
                    styles.triggerText,
                    hasCollapseIcon && iconPosition === 'left' && styles.accordionTriggerContentWithLeftIcon,
                    hasCollapseIcon && iconPosition === 'right' && styles.accordionTriggerContentWithRightIcon,
                  ],
                })}
          >
            {children}
          </String>
        ) : (
          <Box
            {...(isNativeWindInstalled()
              ? {
                  className: cn({
                    'ml-3': hasCollapseIcon && iconPosition === 'left',
                    'mr-3': hasCollapseIcon && iconPosition === 'right',
                  }),
                }
              : {
                  style: [
                    hasCollapseIcon && iconPosition === 'left' && styles.accordionTriggerContentWithLeftIcon,
                    hasCollapseIcon && iconPosition === 'right' && styles.accordionTriggerContentWithRightIcon,
                  ],
                })}
          >
            {children}
          </Box>
        )}
      </Box>

      {hasCollapseIcon && iconPosition === 'right' && (
        <AnimatedBox
          style={{
            transform: [{ rotate: animatedRotation }],
          }}
        >
          <Icon
            name={isOpen ? (collapseIcon as any) || 'angle-down-solid' : (expandIcon as any) || 'angle-down-solid'}
          />
        </AnimatedBox>
      )}
    </>
  );

  // Handle asChild case
  if (asChild) {
    const child = Children.only(children);

    if (isValidElement(child)) {
      const childProps = child.props as any;

      if (isNativeWindInstalled()) {
        return cloneElement(child, {
          ...childProps,
          className: cn(childProps.className, commonProps.className),
          disabled: disabled || childProps.disabled,
          onPress: (e: any) => {
            childProps.onPress?.(e);
            handlePress();
          },
          style: [childProps.style, commonProps.style],
        });
      }
      return cloneElement(child, {
        ...childProps,
        disabled: disabled || childProps.disabled,
        onPress: (e: any) => {
          childProps.onPress?.(e);
          handlePress();
        },
        style: [childProps.style, commonProps.style],
      });
    }
  }

  switch (as) {
    case 'TouchableHighlight':
      return (
        <TouchableHighlight {...commonProps} underlayColor="#F6F7F8">
          {renderContent()}
        </TouchableHighlight>
      );
    case 'Pressable':
      return (
        <Pressable {...commonProps} android_ripple={{ color: '#F6F7F8' }}>
          {renderContent()}
        </Pressable>
      );
    case 'RipplePressable':
      return (
        <RipplePressable onPress={handlePress} className={commonProps.className}>
          {renderContent()}
        </RipplePressable>
      );
    case 'TouchableOpacity':
    default:
      return (
        <TouchableOpacity {...commonProps} activeOpacity={0.7}>
          {renderContent()}
        </TouchableOpacity>
      );
  }
}
