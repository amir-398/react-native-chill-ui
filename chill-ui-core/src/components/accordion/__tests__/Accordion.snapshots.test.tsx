import { render } from '@testing-library/react-native';

import Accordion from '../components/Accordion';
import AccordionItem from '../components/AccordionItem';
import AccordionTrigger from '../components/AccordionTrigger';
import AccordionContent from '../components/AccordionContent';

// Mock des utilitaires hybrid pour Ã©viter les erreurs
jest.mock('../../../utils/hybrid/classNamePropsHandler', () => ({
  classNamePropsHandler: jest.fn(),
}));

jest.mock('../../../utils/hybrid/colorVariantPropsHandler', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('../../../utils/hybrid/propsHandlers', () => ({
  classNameHandler: jest.fn(className => ({ className })),
  styleHandler: jest.fn(style => ({ style: style.defaultStyle })),
}));

// Mock des composants enfants
jest.mock('../../box', () => ({
  Box: ({ children, ...props }: any) => children,
}));

jest.mock('../../string', () => ({
  String: ({ children, ...props }: any) => children,
}));

jest.mock('../../icon', () => ({
  __esModule: true,
  default: ({ name, ...props }: any) => `Icon:${name}`,
}));

jest.mock('../../animatedBox', () => ({
  AnimatedBox: ({ children, ...props }: any) => children,
}));

jest.mock('../../ripplePressable', () => ({
  RipplePressable: ({ children, onPress, ...props }: any) => children,
}));

describe('Accordion Snapshots', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render basic accordion correctly', () => {
    const { toJSON } = render(
      <Accordion>
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render accordion with multiple items correctly', () => {
    const { toJSON } = render(
      <Accordion>
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
        <AccordionItem value="item-2">
          <AccordionTrigger>Question 2</AccordionTrigger>
          <AccordionContent>Answer 2</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render single type accordion correctly', () => {
    const { toJSON } = render(
      <Accordion type="single">
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
        <AccordionItem value="item-2">
          <AccordionTrigger>Question 2</AccordionTrigger>
          <AccordionContent>Answer 2</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render multiple type accordion correctly', () => {
    const { toJSON } = render(
      <Accordion type="multiple">
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
        <AccordionItem value="item-2">
          <AccordionTrigger>Question 2</AccordionTrigger>
          <AccordionContent>Answer 2</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render disabled accordion correctly', () => {
    const { toJSON } = render(
      <Accordion disabled>
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render accordion without icons correctly', () => {
    const { toJSON } = render(
      <Accordion hasCollapseIcon={false}>
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render accordion with left icon position correctly', () => {
    const { toJSON } = render(
      <Accordion iconPosition="left">
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render accordion with custom icons correctly', () => {
    const { toJSON } = render(
      <Accordion expandIcon="chevron-down" collapseIcon="chevron-up">
        <AccordionItem value="item-1">
          <AccordionTrigger>Question 1</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render accordion with disabled item correctly', () => {
    const { toJSON } = render(
      <Accordion>
        <AccordionItem value="item-1" disabled>
          <AccordionTrigger>Question 1 (Disabled)</AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render accordion with asChild trigger correctly', () => {
    const { toJSON } = render(
      <Accordion>
        <AccordionItem value="item-1">
          <AccordionTrigger asChild>
            <div>Custom Trigger</div>
          </AccordionTrigger>
          <AccordionContent>Answer 1</AccordionContent>
        </AccordionItem>
      </Accordion>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
