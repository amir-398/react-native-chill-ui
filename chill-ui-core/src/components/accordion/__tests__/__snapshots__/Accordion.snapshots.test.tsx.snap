// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Snapshots should render accordion with asChild trigger correctly 1`] = `
[
  <div
    className=""
    onPress={[Function]}
  >
    Custom Trigger
  </div>,
  "Answer 1",
]
`;

exports[`Accordion Snapshots should render accordion with custom icons correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1
    Icon:chevron-down
  </View>,
  "Answer 1",
]
`;

exports[`Accordion Snapshots should render accordion with disabled item correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 0.5,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1 (Disabled)
    Icon:angle-down-solid
  </View>,
  "Answer 1",
]
`;

exports[`Accordion Snapshots should render accordion with left icon position correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Icon:angle-down-solid
    Question 1
  </View>,
  "Answer 1",
]
`;

exports[`Accordion Snapshots should render accordion with multiple items correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1
    Icon:angle-down-solid
  </View>,
  "Answer 1",
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 2
    Icon:angle-down-solid
  </View>,
  "Answer 2",
]
`;

exports[`Accordion Snapshots should render accordion without icons correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1
  </View>,
  "Answer 1",
]
`;

exports[`Accordion Snapshots should render basic accordion correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1
    Icon:angle-down-solid
  </View>,
  "Answer 1",
]
`;

exports[`Accordion Snapshots should render disabled accordion correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 0.5,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1
    Icon:angle-down-solid
  </View>,
  "Answer 1",
]
`;

exports[`Accordion Snapshots should render multiple type accordion correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1
    Icon:angle-down-solid
  </View>,
  "Answer 1",
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 2
    Icon:angle-down-solid
  </View>,
  "Answer 2",
]
`;

exports[`Accordion Snapshots should render single type accordion correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 1
    Icon:angle-down-solid
  </View>,
  "Answer 1",
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F0F0F0",
        "borderBottomColor": "#E5E7EB",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    Question 2
    Icon:angle-down-solid
  </View>,
  "Answer 2",
]
`;
